### PR TITLE
r/secret - add `immutable` argument + validations

### PR DESCRIPTION
**Pull request by DrFaust92**
### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
=== RUN   TestAccKubernetesDataSourceSecret_basic
--- PASS: TestAccKubernetesDataSourceSecret_basic (4.34s)
=== RUN   TestAccKubernetesPersistentVolume_cephFsSecretRef
--- PASS: TestAccKubernetesPersistentVolume_cephFsSecretRef (3.46s)
=== RUN   TestAccKubernetesSecret_basic
--- PASS: TestAccKubernetesSecret_basic (11.77s)
=== RUN   TestAccKubernetesSecret_immutable
--- PASS: TestAccKubernetesSecret_immutable (9.84s)
=== RUN   TestAccKubernetesSecret_generatedName
--- PASS: TestAccKubernetesSecret_generatedName (3.24s)
=== RUN   TestAccKubernetesSecret_binaryData
--- PASS: TestAccKubernetesSecret_binaryData (7.09s)
PASS
ok  	github.com/hashicorp/terraform-provider-kubernetes/kubernetes	40.601s
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
resource_kubernetes_secret - add `immutable` argument
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by DrFaust92** [comment]
Rebase and revalidated tests

File: kubernetes/resource_kubernetes_secret.go [hunk 1]
```diff
@@ -22,6 +22,29 @@ func resourceKubernetesSecret() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+			if diff.Id() == "" {
+				return nil
+			}
+
+			// ForceNew if immutable has been set to true
```
Comment (jrhouston) [inline kubernetes/resource_kubernetes_secret.go:30]: @DrFaust92 I've changed the logic here slightly. Before this was only doing a ForceNew on changes to the `immutable` attribute itself (and the tests were only testing this). What we want is to ForceNew on the `data`, `binary_data`, and `immutable` attributes _if_ `immutable` has been previously set to true. 

I've also updated the acceptance test to modify the data field to test this.

**Comment by jrhouston** [comment]
I've also removed the validation that you had [here](https://github.com/hashicorp/terraform-provider-kubernetes/commit/a58daf8c1c621470fb765c462993dc94f7028061#diff-ecab70df659b02330e73d8d51d46bcd3adb460cb77c484794c9c0164b915de7bR71-R80) as the `type` field is actually a free text field in the API with no validation – so I don't see why we would have Terraform add this limitation.

File: kubernetes/resource_kubernetes_secret.go (hunk 1 above)
Comment (DrFaust92) [reply kubernetes/resource_kubernetes_secret.go:30]: Thanks @jrhouston !

**Pull request merged**